packages/node/src/policy-gen/policy-gen-context.js

```diff
@@ -162,55 +173,102 @@ export class PolicyGeneratorContext {
    * @remarks
    * In Endo, the `compartment` is stored as a _string_ `file://` URL; hence the
    * conversion.
-   * @param {string | Required<ModuleDescriptor>} descriptorOrLocation
-   * @returns {string}
+   * @param {ModuleDescriptor} descriptor Module descriptor
+   * @returns {string | undefined}
    * @todo There may be a safer way to do this conversion
    */
-  toPath(descriptorOrLocation) {
-    if (isString(descriptorOrLocation)) {
-      return this.#readPowers.fileURLToPath(new URL(descriptorOrLocation))
-    }
-    const location = this.renames[descriptorOrLocation.compartment]
-    if (!location) {
-      throw new TypeError(
-        `Rename map missing location for compartment "${descriptorOrLocation.compartment}" in compartment ${this.compartment.name}`
+  toPath(descriptor) {
+    // it might have a `compartment` and `module`
+    if (hasValue(descriptor, 'compartment') && hasValue(descriptor, 'module')) {
+      const location = this.renames[descriptor.compartment]
+      if (!location) {
+        throw new TypeError(
+          `Compartment ${hrLabel(this.canonicalName)}: Rename map missing location for referenced compartment ${hrPath(descriptor.compartment)}`
+        )
+      }
+      return this.#readPowers.fileURLToPath(
+        new URL(descriptor.module, location)
       )
     }
-    return this.#readPowers.fileURLToPath(
-      new URL(descriptorOrLocation.module, location)
-    )
   }
 
   /**
-   * Attempts to find the filepath of a specifier
+   * Attempts to resolve the filepath of `specifier` using the current
+   * compartment descriptor's
+   * {@link CompartmentDescriptor.modules module descriptors}.
    *
    * Strips trailing slashes from specifiers, since Endo does not store them
    * with trailing slashes
    *
    * @param {string} specifier
-   * @returns {string | undefined}
+   * @returns {string | null} Filepath, name of builtin, or if `null`, then
+   *   either the module is missing or we just don't need an import map entry
+   *   for it.
    */
-  getFilepath(specifier) {
+  resolve(specifier) {
```

Comment:
Could we rename this to something more unique so it's not this hard to find usage of?

```suggestion
  resolveSpecifier(specifier) {
```

```suggestion
  resolvePathFromSpecifier(specifier) {
```

Reply:
unfortunately I cannot use the suggestion since this is not the only appearance of a reference to this function, but ok.

Reply:
- [x] rename `resolve`